### ISSUE TITLE
sync: 'xml.etree.ElementTree.Element' object has no attribute 'getiterator'

### ISSUE THREAD

**Issue by kierun** [pimutils/vdirsyncer#880]
I ran `vdirsyncer discover calendar` which returned successfully and then `vdirsyncer sync` which did not: I got several `error: Unknown error occurred for calendar/family04493139417728447737@group.calendar.google.com: 'xml.etree.ElementTree.Element' object has no attribute 'getiterator'`.

It might be something I did badly while setting up… 

* Your vdirsyncer version: **vdirsyncer, version 0.16.8**
* If applicable, which server software (and which version) you're using
    * Your Python version: **3.9.0**
    * Your operating system: **Ubuntu 20.04**
* Your config file:
```
[general]
status_path = "~/.vdirsyncer/status/"

[pair calendar]
a = "google_calendar"
b = "local_calendar"
collections = ["from a", "from b"]
metadata = ["color"]

[storage local_calendar]
type = "singlefile"
path = "~/.calendars/%s.ics"

[storage google_calendar]
type = "google_calendar"
token_file = "~/.vdirsyncer/google_token"
client_id = "…"
client_secret = "…"
```

* Use `vdirsyncer -vdebug` for debug output. The output is sensitive, but
  please attach at least the last few lines before the error (if applicable),
  censored as necessary. This is almost always the most useful information.

```
debug: Normalized URL from '/caldav/v2/cln2stbb4dk6ur39chgnig37e9nnas1eeon66obccln68obi5pjmurr7dhiisorfdk%40virtual/events/' to '/caldav/v2/cln2stbb4dk6ur39chgnig37e9nnas1eeon66obccln68obi5pjmurr7dhiisorfdk@virtual/events/'
error: Unknown error occurred for calendar/cln2stbb4dk6ur39chgnig37e9nnas1eeon66obccln68obi5pjmurr7dhiisorfdk@virtual: 'xml.etree.ElementTree.Element' object has no attribute 'getiterator'
error: Use `-vdebug` to see the full traceback.
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/cli/tasks.py", line 62, in sync_collection
debug:     sync.sync(
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/sync/__init__.py", line 134, in sync
debug:     a_nonempty = a_info.prepare_new_status()
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/sync/__init__.py", line 44, in prepare_new_status
debug:     for href, etag in self.storage.list():
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/storage/dav.py", line 792, in list
debug:     yield from DAVStorage.list(self)
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/storage/dav.py", line 635, in list
debug:     for href, etag, _prop in rv:
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/storage/dav.py", line 591, in _parse_prop_responses
debug:     props = _merge_xml(props)
debug:   File "/home/usr/.pyenv/versions/3.9.0/lib/python3.9/site-packages/vdirsyncer/storage/dav.py", line 123, in _merge_xml
debug:     rv.extend(item.getiterator())
error: 5 out of 6 tasks failed.
```

**Comment by danields761** [comment]
Actually, this is blocker for Python >= 3.9, because `xml.etree.ElementTree.Element.getiterator` has been deprecated from version 3.2 and marked for removal in 3.9.

For those folks, who doesn't want to wait untill new release, there is quite simple fix: just replace `rv.extend(item.getiterator())` with `rv.extend(item)` and it will work.

**Comment by mcepl** [comment]
More specifically [this patch](https://github.com/pimutils/vdirsyncer/files/6571360/deprecated_getiterator.patch.txt) fixes the problem.

**Comment by WhyNotHugo** [comment]
This has been fixed in 7577fa21177442aacc2d86640ef28cebf1c4aaef.
